pkg/secrethub/secret.go

```diff
@@ -171,7 +171,7 @@ func (s secretService) Write(path string, data []byte) (*api.SecretVersion, erro
 	}
 
 	key, err := s.client.getSecretKey(secretPath)
-	if api.IsErrNotFound(err) {
+	if err == api.ErrSecretNotFound {
 		return s.client.createSecret(secretPath, data)
 	} else if err == api.ErrNoOKSecretKey {
 		key, err = s.client.createSecretKey(secretPath)
```

Comment:
Can and should the direct error check be avoided in this case?

Reply:
In this case, it's part of the functionality to check more specifically than is something not found; namely, we distinguish between two cases:
- the secret does not exist yet
- the secret exists, but the secret key has to be refreshed.